pkg/controllers/provisioning/scheduling/nodeclaimtemplate.go

```diff
@@ -62,7 +62,7 @@ func (i *NodeClaimTemplate) ToNodeClaim(nodePool *v1beta1.NodePool) *v1beta1.Nod
 	nc := &v1beta1.NodeClaim{
 		ObjectMeta: metav1.ObjectMeta{
 			GenerateName: fmt.Sprintf("%s-", i.NodePoolName),
-			Annotations:  lo.Assign(i.Annotations, map[string]string{v1beta1.NodePoolHashAnnotationKey: nodePool.Hash()}),
+			Annotations:  lo.Assign(i.Annotations, map[string]string{v1beta1.NodePoolHashAnnotationKey: nodePool.Annotations[v1beta1.NodePoolHashAnnotationKey]}),
```

Comment:
What happens here if the NodePool hash hasn't been reconciled yet from this NodePool? I think that we grab these NodePools really early on in the scheduling loop so this may be possible. 

The other thing: What's the necessity of changing this logic? Shouldn't new NodeClaims just use whatever the hashing algorithm is for the current in-memory schema?

Reply:
Removed for this design.